graphql/schema.js

```diff
@@ -206,5 +207,43 @@ input DocumentInput {
   # AST of /article.md
   content: JSON!
 }
+
+interface EmbedInterface {
+  id: ID!
+  url: String!
+}
+
+type Twitter implements EmbedInterface {
+  id: ID!
+  url: String!
+
+  text: String!
+  createdAt: String!
+  userId: String!
+  userName: String!
+  userScreenName: String!
+}
+
+type Youtube implements EmbedInterface {
+  id: ID!
+  url: String!
+
+  createdAt: String!
+  userId: String!
+  userName: String!
+  thumbnail: String!
+}
+
+type Vimeo implements EmbedInterface {
+  id: ID!
+  url: String!
+
+  createdAt: String!
+  userId: String!
+  userName: String!
+  thumbnail: String!
+}
+
+union Embed = Twitter | Youtube | Vimeo
```

Comment:
IMHO `union Embed = TwitterEmbed | YoutubeEmbed | VimeoEmbed` would be better language.